### PR TITLE
Update get delta api to log caller and restrict the requested ops size

### PR DESCRIPTION
**Pull request by shubhi1092**
## Description

- Adding "caller" property to the telemetry to identify internal vs external calls.
- Adding a check for the number of ops requested in getDeltas API call, and return error if more than 2000 (max range is configurable).

### CODE REVIEW

File: server/routerlicious/packages/routerlicious-base/src/alfred/routes/api/deltas.ts [hunk 1]
```diff
@@ -147,6 +148,20 @@ export function create(
...
+				return handleResponse(
+					Promise.reject(
+						new NetworkError(
+							400,
+							`Requested too many ops. Max ops range is: ${getDeltasRequestMaxOpsRange}`,
+						),
+					),
+					response,
+				);
+			}
```
Comment (zhangxin511) [inline server/routerlicious/packages/routerlicious-base/src/alfred/routes/api/deltas.ts:164]: Nit, could we create a middleware like `validateRequestParams` to validate the sn range? Of course, if it leads duplicated code like parsing the `from` `to`, then no need to be that picky.

File: server/routerlicious/packages/lambdas/src/scribe/summaryWriter.ts [hunk 2]
```diff
@@ -713,7 +713,14 @@ export class SummaryWriter implements ISummaryWriter {
 		let logTail: ISequencedDocumentMessage[] = [];
 
 		if (this.getDeltasViaAlfred) {
-			logTail = await this.deltaService.getDeltas("", this.tenantId, this.documentId, gt, lt);
+			logTail = await this.deltaService.getDeltas(
+				"",
+				this.tenantId,
+				this.documentId,
+				gt,
+				lt,
```
Comment (zhangxin511) [inline server/routerlicious/packages/lambdas/src/scribe/summaryWriter.ts:721]: We may need to do a check here, if `gt` and `lt` range is larger than `getDeltasRequestMaxOpsRange`, we break the request into chunks with max range of `getDeltasRequestMaxOpsRange`

File: server/routerlicious/packages/lambdas/src/scribe/pendingMessageReader.ts [hunk 3]
```diff
@@ -26,6 +26,7 @@ export class PendingMessageReader implements IPendingMessageReader {
 			this.documentId,
 			from - 1,
 			to + 1,
```
[the remark this answers is not in the document]
Comment (shubhi1092) [reply server/routerlicious/packages/lambdas/src/scribe/pendingMessageReader.ts:28]: Added some checks in server/routerlicious/packages/routerlicious-base/src/alfred/routes/api/deltas.ts to use default values if from / to are undefined.

File: server/routerlicious/packages/lambdas/src/scribe/summaryWriter.ts (hunk 2 above)
Comment (shubhi1092) [reply server/routerlicious/packages/lambdas/src/scribe/summaryWriter.ts:721]: good point, I updated the logic for that in deltaManager.ts

**Pull request closed without merging**